### ISSUE TITLE
False positive `nested-min-max` for nested lists

### ISSUE THREAD

**Issue by udifuchs** [pylint-dev/pylint#9307]
### Bug description

```python
"""False positive `nested-min-max` for nested lists"""
a = [[1, 2, 3], [4, 5, 6]]
print(max(max(a)))  # Output: 6
print(max(*a))  # Output: [4, 5, 6]
```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
************* Module a
a.py:3:6: W3301: Do not use nested call of 'max'; it's possible to do 'max(*a)' instead (nested-min-max)
```

### Expected behavior

Pylint gives wrong advice. If `a` is a nest list, `max(max(a))` is not equivalent  `max(*a)`.

### Pylint version

```shell
pylint 3.0.3
astroid 3.0.2
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
```

**Comment by jacobtylerwalls** [comment]
I wonder if we can recommend itertools.chain.from_iterable() here instead. Thanks for the clear report.

**Comment by Pierre-Sassoulas** [comment]
Isnt't max(max(a)) more elegant ? Should we consider this a false positive ?

**Comment by jacobtylerwalls** [comment]
I haven't benchmarked itertools but I would assume it's better than iterating twice. chain.from_iterable is marginally uglier, but I would guess that when people prefer explicitness over performance they'd probably disable the whole check. Thoughts?

**Comment by Pierre-Sassoulas** [comment]
Ha sure, sometime I don't remember or it's unclear what the goal of a check is. perfs and readability are sometime incompatible.
